pkg/service/auth.go

```diff
@@ -237,7 +236,10 @@ func (a *AuthService) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 // and returns status `OK` or not `OK`.
 func (a *AuthService) Check(parentContext gocontext.Context, req *envoy_auth.CheckRequest) (*envoy_auth.CheckResponse, error) {
 	requestData := req.Attributes.Request.Http
-	requestId := requestData.GetId()
+	requestId := ensureRequestId(requestData.GetId())
+	if requestData.GetId() == "" {
+		req.Attributes.Request.Http.Id = requestId
+	}
```

Comment:
Maybe we don't need the conditional:

```go
req.Attributes.Request.Http.Id = ensureRequestId(requestData.GetId())
```

Or if the assignment might be costly:

```go
	if requestData.GetId() == "" {
		req.Attributes.Request.Http.Id = ensureRequestId(requestData.GetId())
	}
```